### PR TITLE
feat(common-utils): adding new settings module for defining attributes and assiging values

### PR DESCRIPTION
**Pull request by braunreyes**
## Goal
What changed? What is the business/product goal?

Adding a new `settings` module to `common-utils`.  This will allow for a repeatable process for creating and populating user defined settings for Prefect flows and helper libs.

## Implementation Decisions

This work is based on [Pydantic Settings Management](https://docs.pydantic.dev/latest/usage/settings/).

The idea is that you can create Pydantic models for the different sets of settings that you need using the `NestedSettings` class and then populate them via the `Settings` class using environment variables that start with `DF_CONFIG_`.  Here is the envar format:

`DF_CONFIG_<Settings attribute name>__<NestedSettings attribute name>=<value>`

Since the existence of a nested attribute requires the definition of a NestedSettings mode, the whole thing can be set with a JSON string:

`DF_CONFIG_<Settings attribute name>='{"nested_attribute_name": "value"}'`

In cases, where a NestedSettings model is not needed, single values work as well:
 
`DF_CONFIG_<Settings attribute name>=<value>`

Here is an example of usage:

```python
from common.settings import NestedSettings, Settings
from prefect_snowflake import SnowflakeConnector, SnowflakeCredentials
from pydantic import SecretStr


class SnowflakeCredSettings(NestedSettings):
    account: str
    user: str
    password: SecretStr


class SnowflakeDbSettings(NestedSettings):
    database: str
    warehouse: str
    schema_name: str


class SnowflakeSettings(Settings):
    snowflake_credentials: SnowflakeCredSettings
    snowflake_connector: SnowflakeDbSettings


settings = SnowflakeSettings()  # type: ignore

snowflake_credentials = SnowflakeCredentials(**settings.snowflake_credentials.dict())
snowflake_connector = SnowflakeConnector(
    credentials=snowflake_credentials,
    schema=settings.snowflake_connector.schema_name,  # schema is reserved word in Pydantic  # noqa: E501
    **settings.snowflake_connector.dict()
)

if __name__ == "__main__":
    print(snowflake_credentials.dict())
    print(snowflake_connector.dict())
```

Here are the results:

image

## Deployment steps
- [N/A] Database migrations?
- [N/A] Secrets?
- [N/A] **FOR POCKET DEVELOPERS ONLY** - Post in [\#changelog](https://pocket.slack.com/archives/C0Q4UFMDZ):
> Write a #changelog message using our [best practices](https://docs.google.com/document/d/1oEt8Mtkp-6Xz9S2zaNX1EIXfQIEDN2JpY0diuPc1HZc/edit) if this PR (potentially) impacts users. Describe the 'why' and 'what'. @mention relevant teams. Link to this PR.

## References

JIRA ticket:
[DIS-630](https://getpocket.atlassian.net/browse/DIS-630)

Documentation:
https://getpocket.atlassian.net/wiki/spaces/PE/pages/2956492928/Advanced+Topics

[DIS-630]: https://getpocket.atlassian.net/browse/DIS-630?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: common-utils/poetry.lock [hunk 1]
```diff
@@ -354,89 +354,19 @@ pycparser = "*"
 
 [[package]]
 name = "charset-normalizer"
-version = "3.1.0"
+version = "2.1.1"
```
Comment (mmiermans) [inline common-utils/poetry.lock:357]: Just curious: why did charset-normalizer go to a lower version? I'm guessing these changes were triggered by a version change in pyproject.toml?

File: common-utils/pyproject.toml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = "common-utils"
-version = "0.1.4"
+version = "0.2.0"
```
Comment (mmiermans) [inline common-utils/pyproject.toml:3]: Just curious: why did you decide to do a minor version increase instead of a patch?

**Review by mmiermans — APPROVED** [review]
:+1: Sweet and simple. Looking forward to try this with Snowflake and GCP.

File: common-utils/poetry.lock (hunk 1 above)
Comment (braunreyes) [reply common-utils/poetry.lock:357]: oh, I installed `prefect-snowflake` to create the example and then removed it.  Looks like it downgraded to version that supported both `requests` and `prefect-snowflake`. Then left it when I removed the `prefect-snowflake` lib cause it was still within the version ranges.

File: common-utils/pyproject.toml (hunk 2 above)
Comment (braunreyes) [reply common-utils/pyproject.toml:3]: this was because of the new module, so I was thinking new feature = minor and fixes = patch

**Pull request merged**